references/cuegen/tag_test.go

```diff
@@ -132,8 +132,8 @@ func TestParseExtTag(t *testing.T) {
 		{"bool_and_kv", "key1;key2:value2", map[string]string{"key1": "", "key2": "value2"}},
 		{"kv_and_bool", "key1:value1;key2", map[string]string{"key1": "value1", "key2": ""}},
 		{"multi_kv_and_bool", "key1:value1;key2:value2;key3", map[string]string{"key1": "value1", "key2": "value2", "key3": ""}},
-		{"escape", "key1:\"value\"", map[string]string{"key1": `"value"`}},
-		{"escape_2", "key1:value1\\;vv", map[string]string{"key1": "value1;vv"}},
+		{"escape_value", "key1:value\\;1", map[string]string{"key1": "value;1"}},
```

Comment:
1. Suggestion here, and this is not the most required but made it easier to understand.
```suggestion
		{"escape_value", `key1:value\;1`, map[string]string{"key1": "value;1"}},
```
2. Does the escape only support the separator `;`? How about `,`. Will it detect `\\` means `\`? 
3. Let's add more test about edge case. For example, 
	1. `\` is the last character in string.
	2. `\` followed by a character which can't be escaped.

Reply:
Thanks for review, bugs were found in edge cases and I have fixed it.

Currently only the escape for `;` is supported, other escapes may not be necessary in real world scenarios, so we can add it if someone really needs it...🤔

Reply:
I see. Now if `\` is not followed by a `;` then it means `\` literally. I think other special characters like `,` in this case will appears in the future sooner or later. Why don't we support it instead of leaving it to be fixed.

Reply:
It's supported now: 1d43b31bb19a3c8d5f7344ae30b194bec6aba306

There are many new tests, which need to be review.